deepchem/models/callbacks.py

```diff
@@ -51,6 +61,14 @@ def __init__(self,
       if True, the best model is considered to be the one that minimizes the
       validation metric.  If False, the best model is considered to be the one
       that maximizes it.
+    early_stop_metric: str/int, default None
+      Indicates whether early stopping needs to be applied. If int, the index 
+      of the metric to use when checking for early stopping. If str, the loss 
+      on validation dataset is computed.
+    delta: float, default 0.01
+      Threshold for measuring the change in metric, to focus only on significant changes.
+    patience: int, default 0,
+      Number of steps with no improvement after which early stopping is enforced.
```

Comment:
On both these options, how about adding the following sentence?  "If early_stop_metric is None, this is ignored."

The descriptions could also be a little clearer.  For `delta` maybe add the sentence, "Early stopping will occur when the metric improves by less than this amount."  And for `patience` say, "Number of successive steps..." to clarify that it's counting successive steps, not total steps.